### PR TITLE
testing/ctorrent-dnh: new aport

### PR DESCRIPTION
**Pull request by w1r0x**
http://www.rahul.net/dholmes/ctorrent/
Enhanced CTorrent is a BitTorrent client for unix-like environments

### CODE REVIEW

File: testing/ctorrent-dnh/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Contributor: Denis Ryabyy <vv1r0x@gmail.com> 
+# Maintainer: Denis Ryabyy <vv1r0x@gmail.com>
+pkgname=ctorrent-dnh
+pkgver=3.3.2
+pkgrel=3
```
Comment (jirutka) [inline testing/ctorrent-dnh/APKBUILD]: It’s a new aport, so pkgrel should be 0.

File: testing/ctorrent-dnh/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          msg "Applying ${i}"
+          patch -p1 -i $i || return 1
+        done
+        ./configure --prefix $pkgdir/usr  
+}
+
+build() {
+        cd $_builddir
+        make && make install
+}
```
Comment (jirutka) [inline testing/ctorrent-dnh/APKBUILD]: Please use tabs for indentation in APKBUILDs.

File: testing/ctorrent-dnh/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+pkgname=ctorrent-dnh
+pkgver=3.3.2
+pkgrel=3
+pkgdesc="Enhanced CTorrent is a BitTorrent client for unix-like environments"
+url="http://www.rahul.net/dholmes/ctorrent/"
+arch="all"
+license="GPL2"
+
+source="${pkgname}${pkgver}.tar.gz::https://sourceforge.net/projects/dtorrent/files/dtorrent/${pkgver}/${pkgname}${pkgver}.tar.gz/download"
+_builddir="$srcdir"/${pkgname}${pkgver}
```
Comment (jirutka) [inline testing/ctorrent-dnh/APKBUILD]: `_builddir` is old convention, it should be named `builddir`. This variable is used e.g. by `default_prepare`.

File: testing/ctorrent-dnh/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+prepare() {
+        cd $_builddir
+        for i in ../../*.patch; do
+          msg "Applying ${i}"
+          patch -p1 -i $i || return 1
+        done
```
Comment (jirutka) [inline testing/ctorrent-dnh/APKBUILD]: This is not needed, just call `default_prepare` here, it applies patches.

File: testing/ctorrent-dnh/APKBUILD (hunk 1 above)
Comment (w1r0x) [inline testing/ctorrent-dnh/APKBUILD]: fixed

File: testing/ctorrent-dnh/APKBUILD (hunk 2 above)
Comment (w1r0x) [inline testing/ctorrent-dnh/APKBUILD]: fixed

File: testing/ctorrent-dnh/APKBUILD (hunk 3 above)
Comment (w1r0x) [inline testing/ctorrent-dnh/APKBUILD]: fixed

File: testing/ctorrent-dnh/APKBUILD (hunk 4 above)
Comment (w1r0x) [inline testing/ctorrent-dnh/APKBUILD]: fixed

File: testing/ctorrent-dnh/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+prepare() {
+	default_prepare
+	./configure --prefix $pkgdir/usr
+}
+
+build() {
+	cd "$builddir"
+	make
+	make install
```
Comment (jirutka) [inline testing/ctorrent-dnh/APKBUILD]: `make install` must be in `package()` function, that is mandatory for abuilds. Have you removed it (why?), or it wasn’t here even before and I overlooked it?

Comment (w1r0x) [inline testing/ctorrent-dnh/APKBUILD]: Somehow when it in package() function binary not including into the package. But it in pkg dir if I run steps manually with abuild. I'll give you more detailed answer when be near my pc.

Comment (jirutka) [inline testing/ctorrent-dnh/APKBUILD]: You have to also declare DESTDIR (or similar variable) to specify destination directory `$pkgdir`.

**Comment by w1r0x** [comment]
>./configure --prefix $pkgdir/usr
Is it not the one?

**Comment by w1r0x** [comment]
```
package () {
	cd "$builddir"
	make install
	cd "$pkgdir"
	find
}
```
I add this construction as a test.

First abuild on clean system is successfull but every other time when I'm doing
`rm -rf /home/w1r0x/packages/testing/x86_64 && abuild` a get this:
```
>>> ctorrent-dnh: Entering fakeroot...
make[1]: Entering directory '/home/w1r0x/aports/testing/ctorrent-dnh/src/ctorrent-dnh3.3.2'
make[1]: Nothing to be done for 'install-data-am'.
test -z "/home/w1r0x/aports/testing/ctorrent-dnh/pkg/ctorrent-dnh/usr/bin" || /bin/sh ./mkinstalldirs "/home/w1r0x/aports/testing/ctorrent-dnh/pkg/ctorrent-dnh/usr/bin"
mkdir /home/w1r0x/aports/testing/ctorrent-dnh/pkg
mkdir /home/w1r0x/aports/testing/ctorrent-dnh/pkg/ctorrent-dnh
mkdir /home/w1r0x/aports/testing/ctorrent-dnh/pkg/ctorrent-dnh/usr
mkdir /home/w1r0x/aports/testing/ctorrent-dnh/pkg/ctorrent-dnh/usr/bin
  /usr/bin/install -c 'ctorrent' '/home/w1r0x/aports/testing/ctorrent-dnh/pkg/ctorrent-dnh/usr/bin/ctorrent'
make[1]: Leaving directory '/home/w1r0x/aports/testing/ctorrent-dnh/src/ctorrent-dnh3.3.2'
.
./usr
./usr/bin
./usr/bin/ctorrent
>>> ctorrent-dnh*: Running postcheck for ctorrent-dnh
>>> ctorrent-dnh*: Preparing package ctorrent-dnh...
>>> ctorrent-dnh*: Stripping binaries
>>> ctorrent-dnh*: Scanning shared objects
>>> ctorrent-dnh*: Tracing dependencies...
     so:libc.musl-x86_64.so.1
     so:libcrypto.so.1.0.0
     so:libgcc_s.so.1
     so:libstdc++.so.6
>>> ctorrent-dnh*: Package size: 0.0 
>>> ctorrent-dnh*: Compressing data...
>>> ctorrent-dnh*: Create checksum...
>>> ctorrent-dnh*: Create ctorrent-dnh-3.3.2-r0.apk
>>> ctorrent-dnh: Cleaning up srcdir
>>> ctorrent-dnh: Cleaning up pkgdir
>>> ctorrent-dnh: Updating the cached abuild repository index...
```

>Package size: 0.0 

But `find` in `$pkgdir` shows that binary is there:
```
.
./usr
./usr/bin
./usr/bin/ctorrent

```

Is this OK?

**Pull request closed without merging**